src/components/dashboard/Orders.tsx

```diff
@@ -1,93 +1,283 @@
+
+    const [searchField, setSearchField] = useState<string>("");
+    const [searchResults, setSearchResults] = useState<SearchResultState>([]);
+
+    const [isFilePicked, setIsFilePicked] = useState(false);
+
+
+    //* Datagrid Table
+    const [rows, setRows] = useState<rowState>([]);
+    const [columns, setColumns] = useState<GridColDef[]>([]);
+
+    //* Search area
+    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setSearchField((e.target.value).toLowerCase());
+        console.log(searchField);
+    };
+
+    //!Helper function to get header (for Dynamic Header)
+    // function getSheetHeaders(sheet: XLSX.WorkSheet) {
+    //     const headerRegex = new RegExp("^([A-Za-z]+)1='(.*)$");
+
+    //     const cells = XLSX.utils.sheet_to_formulae(sheet);
+    //     return cells.filter(item => headerRegex.test(item)).map(item => item.split("='")[1]);
+    // }
+
+    //* upload file
+    const changeHandler = async (e: ChangeEvent<HTMLInputElement>) => {
+
+        setIsFilePicked(true);
+
+        if (!e.target.files) {
+            console.error("Select a file");
+            return;
+        }
+
+        const file = e.target.files[0];
+
+        const data = await file.arrayBuffer();
+
+        const wb = XLSX.read(data);
+
+        const ws = wb.Sheets[wb.SheetNames[0]];
+
+
+        const rows = XLSX.utils.sheet_to_json(ws);
+     
+        if(Array.isArray(rows)){
+            setRows(rows);
+
+            //! in case of using dynamic Header
+            // setRows(rows.slice(1))
+        }
+
+ 
+
+        //! Dynamic Header
+        // const range = XLSX.utils.decode_range(ws["!ref"] || "A1");
+        // const columns = Array.from({ length: range.e.c + 1 }, (_, i) => ({
+        //     field: String(i), // MUIDG will access row["0"], row["1"], etc
+        //     headerName: getSheetHeaders(ws)[i], // the column labels
+        //     editable: true, // enable cell editing
+        //     headerClassName: "super-app-theme--header",
+        // }));
+
+        //static Header
+        const columns: GridColDef[] = [
+            { field: "id", headerName: "No.", width: 90 , align:"center",},
+            {
+                field: "fullName",
+                headerName: "Name",
+                width: 200,
+                align:"center",
+
+            },
+            {
+                field: "totalBoxes",
+                headerName: "Boxes Ordered",
+                width: 150,
+                align:"center",
+
+            },
+            {
+                field: "totalAmount",
+                headerName: "Total Amount",
+                type: "number",
+                width: 150,
+                align:"center",
+            },
+            {
+                field: "isPaid",
+                headerName: "Paid",
+                width: 150,
+                align:"center",
+                renderCell: (params) => (
+                    <span style={params.value==="Yes" || params.value==="yes"?{ 
```

Comment:
This ternary operator formatting is a bit awkward. Please use 

condition 
? 
{ true case }
: 
{ false case }